extensions/amp-base-carousel/1.0/amp-base-carousel.js

```diff
@@ -155,13 +156,20 @@ AmpBaseCarousel['useContexts'] = [CarouselContextProp];
  * @private
  */
 function fireSlideChangeEvent(win, el, index, trust) {
-  const name = 'slideChange';
+  const eventName = 'slideChange';
+  const data = dict({'index': index});
   const slideChangeEvent = createCustomEvent(
     win,
-    `amp-base-carousel.${name}`,
-    dict({'index': index})
+    `amp-base-carousel.${eventName}`,
```

Comment:
To confirm, the use is to do:

```
element.addEventListener('amp-base-carousel.slidechange', () => {
  ...
});
```

?

Couple of things to discuss:
1. I think most of time we've so far sent simpler event names. E.g. "eventx", not "amp-foo.eventx". Would be good to figure out which form is better and safer.
2. Does a complicated event name like "amp-foo.eventx" work in reality?
3. Are event names case-insensitive? I.e. does it matter that it's "slideChange" above vs "slidechange"?

Reply:
The use is to do 
```
carousel.addEventListener('slideChange', () => {
  ...
});
```

The `amp-base-carousel.slideChange` name format is for the AMP event, which AFAIK are in this format. But the event being fired in the DOM is a few lines later:
https://github.com/ampproject/amphtml/blob/a19d199431697c3cd80ac49cdcd6d236aeb05be3/extensions/amp-base-carousel/1.0/amp-base-carousel.js#L172

I believe this covers 1/2. As for 3, `slideChange` works and `slidechange` does not - is there a way to make events case insensitive?

Reply:
IMHO, for now let's dispatch "slideChange" only. And we'll follow up with a discussion on whether we want to switch to all lower-case.